manifests/init.pp

```diff
@@ -207,6 +215,37 @@
     order   => '01',
   }
 
+  # check if this is really a bool
+  validate_bool($use_ldap)
+  if $use_ldap {
+    unless ($ldap_method in ['dynamic', 'static']) {
+      fail('$ldap_method must be dynamic or static')
+    }
+    if ($ldap_password == '') {
+      fail('you must set $ldap_password')
+    }
+    unless (is_integer($ldap_port)) {
+      fail('$ldap_port must be an integer')
+    }
+    if ($ldap_server == '') {
+      fail('you must set $ldap_server')
+    }
+    if ($ldap_username == '') {
+      fail('you must set $ldap_username')
+    }
+    if ($ldap_base_dn == '') {
+      fail('you must set $ldap_username')
+    }
+    if ($ldap_debug_file == '') {
```

Comment:
as mentioned in the commit message, validate_absolute_path needs a path to a directory, not to a filename so I removed it after I first added it.